LSPS1/README.md

```diff
@@ -0,0 +1,503 @@
+# LSPS1 Channel Request
+
+| Name    	| channel_request               |
+|---------	|------------------------------	|
+| Version 	| 2                             |
+| Status    | For Implementation            |
+
+
+## Motivation
+
+The goal of this specification is to provide a standardized LSP API for wallets to purchase a channel from the LSP directly.
+
+## Definitions
+
+### Data types
+
+This specification uses data types defined in [LSPS0 Common Schemas](LSPS0/common-schemas.md). The types are referenced in the format of `LSPS0.{type}`.
+
+### Actors
+
+`LSP` is the API provider. `Client` is the client of the API.
+
+### Channel sides
+
+`client_balance` are the funds on the client side. `lsp_balance` are the funds on the LSP side.
+
+### Overprovisioning
+
+The LSP is allowed to overprovision channels/onchain-payments/onchain-fees as long as it benefits the client.
+
+> **Rationale** The lsp may need to "bin" UTXOs.
+
+### Errors
+
+LSP will return errors according to the JSONRPC 2.0 specification (see [LSPS0 Error Handling](https://github.com/BitcoinAndLightningLayerSpecs/lsp/tree/main/LSPS0#error-handling)).
+
+**LSP**
+- MAY return an `code` that is not listed above.
+
+**Client**
+- MUST be able to handle an `code` that is not listed
+  above.
+  - SHOULD report an unrecognized error code simply as
+    "unrecognized error code".
+  - **Rationale** This allows clients written for older versions
+    of this specification to work with LSPs written for newer
+    versions.
+- SHOULD NOT display an unrecognized `code`, `message` or `data` to the user.
+  **Rationale** Unrecognized `code`s may be misinterpreted
+  or misunderstood by users.
+
+Any error MAY include a `message` field that is a human-readable string. The `message` field is intended to provide a short description of the error. The `message` field is not intended to be parsed by the client or shown to the user. It is intended as a developer hint to help the client developer debug the error.
+
+## Overview
+
+1. Client calls `lsps1.info` to get the LSP's API version and options.
+2. Client calls `lsps1.create_order` to create an order.
+3. Client pays the order either onchain or offchain.
+4. LSP opens the channel as soon as they payment is confirmed.
+5. LSP refunds the client in case the channel open failed.
+
+## API
+
+### 1. lsps1.info
+
+| JSONRPC Method | lsps1.info |
+|--------        |------------|
+| Idempotent     | Yes        |
+
+
+`lsps1.info` is the entrypoint for each client using the api. It lists the supported versions of the api and all options in a dictionary.
+The client MUST call `lsps1.info` first.
+
+**Request** No parameters needed.
+
+**Response** 
+
+```JSON
+{
+  "versions": [2],
+  "website": "http://example.com/contact",
+  "options": {
+      "minimum_depth_channel": 0,
+      "minimum_onchain_payment_confirmations": 1,
+      "supports_zero_channel_reserve": true,
+      "min_required_onchain_sat": null,
```

Comment:
Will change to `min_onchain_payment_size_sat`.

>Also, I'm not quite sure why this field is necessary, couldn't the LSP simply leave out the onchain funding address in case the client creates an order smaller than this amount?

Some thoughts about `info()`:
- This data should not change that often. Explorer like Amboss should be able to list it on their website.
- Reading `min_onchain_payment_size_sat` will prevent users from creating orders. This reduces the servers requirements to watch onchain addresses and create lightning invoices.

I guess I need to add this rationale to the spec.